hq/test/logic/SnykDisplayTest.scala

```diff
@@ -0,0 +1,209 @@
+    organisationId.isFailedAttempt shouldBe true
+    organisationId.getFailedAttempt.get.failures.head.friendlyMessage shouldBe "Could not read Snyk response ({\"toughluck\": \"no use\"})"
+  }
+
+  "find project id list" in {
+    val projects = SnykDisplay.getProjectIdList(SnykDisplayTest.mockGoodProjectResponse)
+    projects.value().find(p => p.name == "project1").isEmpty shouldBe false
+    projects.value().find(p => p.name == "project2").isEmpty shouldBe false
+  }
+
+  "fail to find project id list (nice)" in {
+    val projects = SnykDisplay.getProjectIdList(SnykDisplayTest.mockBadResponseWithMessage)
+    projects.isFailedAttempt shouldBe true
+    projects.getFailedAttempt.get.failures.head.friendlyMessage shouldBe "Could not read Snyk response (some nice error)"
+  }
+
+  "fail to find project id list (not nice)" in {
+    val projects = SnykDisplay.getProjectIdList(SnykDisplayTest.mockBadResponseWithoutMessage)
+    projects.isFailedAttempt shouldBe true
+    projects.getFailedAttempt.get.failures.head.friendlyMessage shouldBe "Could not read Snyk response ({\"toughluck\": \"no use\"})"
+  }
+
+  "find empty vulnerability list" in {
+    val projects = SnykDisplay.parseProjectVulnerabilities(List(SnykDisplayTest.mockGoodAndNotVulnerableResponse))
+    projects.value().head.ok shouldBe true
+  }
+
+  "find non-empty vulnerability list" in {
+    val projects = SnykDisplay.parseProjectVulnerabilities(List(SnykDisplayTest.mockGoodButVulnerableResponse))
+    projects.value().head.ok shouldBe false
+    projects.value().head.high shouldBe 1
+    projects.value().head.medium shouldBe 0
+    projects.value().head.low shouldBe 0
+  }
+
+  "fail to find vulnerability list (nice)" in {
+    val projects = SnykDisplay.parseProjectVulnerabilities(List(SnykDisplayTest.mockBadResponseWithMessage))
+    projects.isFailedAttempt() shouldBe true
+    projects.getFailedAttempt.get.failures.head.friendlyMessage shouldBe "Could not read Snyk response (some nice error)"
+  }
+
+  "fail to find vulnerability list (not nice)" in {
+    val projects = SnykDisplay.parseProjectVulnerabilities(List(SnykDisplayTest.mockBadResponseWithoutMessage))
+    projects.isFailedAttempt() shouldBe true
+    projects.getFailedAttempt.get.failures.head.friendlyMessage shouldBe "Could not read Snyk response ({\"toughluck\": \"no use\"})"
+  }
+
+  "label projects" in {
+    val results = SnykDisplay.labelProjects(SnykDisplayTest.goodProjects, SnykDisplayTest.goodVulnerabilities)
+    results.head.id shouldBe "id1"
+    results.head.name shouldBe "name1"
+    results.tail.head.id shouldBe "id2"
+    results.tail.head.name shouldBe "name2"
+  }
+
+}
+
+object SnykDisplayTest  {
+
+  val mockBadResponseWithMessage = s"""{\"error\": \"some nice error\"}"""
+
+  val mockBadResponseWithoutMessage = s"""{\"toughluck\": \"no use\"}"""
+
+  val mockGoodOrganisationResponse =
```

Comment:
A nice technique is to put test JSON into the test resources directory and load it from there in your tests. Escaping JSON is not fun and it's a lot of noise for the test files.

Also, I may be mistaken but it looks like your " escapes aren't required? Within a `"""`, `"` is a literal.